### PR TITLE
Adding support interval (SI)

### PR DESCRIPTION
**Pull request by strengejacke**
https://github.com/easystats/bayestestR/issues/266

### CODE REVIEW

**Comment by strengejacke** [comment]
I think the tests still cause check failures?

File: R/bayesfactor_parameters.R [hunk 1]
```diff
@@ -209,22 +209,17 @@ bayesfactor_parameters.stanreg <- function(posterior,
...
-  if (is.null(prior)) {
-    prior <- posterior
-  }
 
-  prior <- .update_to_priors(prior, verbose = verbose)
-  prior <- insight::get_parameters(prior, effects = effects, component = component)
-  posterior <- insight::get_parameters(posterior, effects = effects, component = component)
+  samps <- .clean_priors_and_posteriors(posterior, prior,
+                                        verbose = verbose,
+                                        effects = effects, component = component)
```
Comment (strengejacke) [inline R/bayesfactor_parameters.R]: Not sure what these changes all do, but I guess you know what you did? ;-)
If everything is working, I'd say these changes are ok, ready to merge.

File: R/bayesfactor_restricted.R [hunk 2]
```diff
@@ -151,34 +147,16 @@ bayesfactor_restricted.stanreg <- function(posterior, hypothesis, prior = NULL,
...
-    prior <- .update_to_priors(prior, verbose = verbose)
-    prior <- emmeans::ref_grid(prior)
-    prior <- prior@post.beta
-    prior <- stats::update(posterior, post.beta = prior)
-  }
-
-  prior <- as.data.frame(as.matrix(emmeans::as.mcmc.emmGrid(prior, names = FALSE)))
-  posterior <- as.data.frame(as.matrix(emmeans::as.mcmc.emmGrid(posterior, names = FALSE)))
+  samps <- .clean_priors_and_posteriors(posterior, prior,
+                                        verbose = verbose)
```
Comment (strengejacke) [inline R/bayesfactor_restricted.R]: Same here, so comment above.

File: R/plot.R [hunk 3]
```diff
@@ -60,6 +60,14 @@ plot.bayestestR_eti <- function(x, ...) {
...
 }
 
+#' @export
+plot.bayestestR_eti <- function(x, ...) {
+  # if (!requireNamespace("see", quietly = TRUE)) {
+  #   stop("Package 'see' needed to plot credible intervals. Please install it by running `install.packages('see')`.")
+  # }
+  stop("No plotting method for SIs (yet)..")
+}
+
```
Comment (strengejacke) [inline R/plot.R]: I would already activate plotting, the implementation in _see_ should be fairly easy, or not?

File: vignettes/credible_interval.Rmd [hunk 4]
```diff
@@ -121,4 +121,35 @@ This is not the case for skewed distributions. Indeed, it is possible that param
...
+            data = estimate_density(posterior, extend = TRUE)) +
+  # The prior
+  geom_area(color = "black", fill = NA, size = 1, linetype = "dashed",
+            data = estimate_density(prior, extend = TRUE)) +
+  # BF = 1 SI in blue
+  geom_vline(xintercept=si_1$CI_low, color="royalblue", size=1) +
+  geom_vline(xintercept=si_1$CI_high, color="royalblue", size=1) +
+  # BF = 3 SI in red
+  geom_vline(xintercept=si_3$CI_low, color="red", size=1) +
+  geom_vline(xintercept=si_3$CI_high, color="red", size=1)
```
Comment (strengejacke) [inline vignettes/credible_interval.Rmd]: This could be the base for `see::plot.si()`, right?

Comment (mattansb) [reply vignettes/credible_interval.Rmd]: I think the plotting method will be based on the method for `bf_parameters`, but the end result will be similar, yes (:

File: R/bayesfactor_parameters.R (hunk 1 above)
Comment (mattansb) [reply R/bayesfactor_parameters.R]: I hope I know what I'm doing ^_^

This and the following changes are just to reduce duplicated code the repeats in `bf_parametes` `bf_restrict` and `si`.

File: R/plot.R (hunk 3 above)
Comment (mattansb) [reply R/plot.R]: I'm not in the right head space for this just yet (still with a bit of the flu)... Hope to get to the plotting for the next PR from dev.

Also, Daniel, can you help with the last issue in #266 ? That of `ci(...,method = "SI")`?

**Comment by mattansb** [comment]
No idea why Travis is failing (after adding `skip_on_travis` no less...) - locally it passes... I think this can be merged, unless @strengejacke / @DominiqueMakowski you want to resolve the SI method for `ci()` first?

**Comment by strengejacke** [comment]
Yes, let us merge this pr

**Comment by DominiqueMakowski** [comment]
@strengejacke doesnt trust Travis anyway so :grin:

**Comment by strengejacke** [comment]
You're approaching new year! About 8 hours left according to easystats time...

**Pull request merged**